processor/src/test/java/org/mapstruct/ap/testutil/runner/CompilationCache.java

```diff
@@ -0,0 +1,54 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.testutil.runner;
+
+import org.mapstruct.ap.testutil.compilation.model.CompilationOutcomeDescriptor;
+
+/**
+ * Remembers the last {@link CompilationRequest} together with the outcome of that request. Consecutive tests that
+ * request the same compilation can reuse the results from this holder.
+ *
+ * @author Andreas Gudian
+ */
+class CompilationCache {
```

Comment:
The cache is still needed to avoid recompilation for consecutive test methods. That's a major speedup for our tests, and test classes where each method brings their own set of classes to compile are noticably slower.